registry/finders.go

```diff
@@ -113,28 +122,125 @@ func FindAppAttachment(c *Space, appSlug, filename string, channel Channel) (*At
-	if err != nil {
-		return nil, err
+
+	// First, we try to get the attachment from the global asset database.
+	ver, err := FindVersion(c, appSlug, version)
+
+	md5Sum, ok := ver.AttachmentReferences[filename]
+	if ok {
+		contentBuffer, headers, err = asset.AssetStore.FS.GetAsset(md5Sum)
+		fileContent, err = ioutil.ReadAll(contentBuffer)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// If we cannot find it, we try from the local database as a fallback
+		prefix := GetPrefixOrDefault(c)
+		headers, err = sc.ObjectGet(prefix, fp, contentBuffer, false, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		fileContent = contentBuffer.Bytes()
+
+		// We are going to move the asset to the global database and remove it
+		// from the local database for the next time
+		go func() {
+			err = MoveAssetToGlobalDatabase(c, ver, bytes.NewReader(fileContent), filename, contentType)
+			if err != nil {
+				log := logrus.WithFields(logrus.Fields{
+					"nspace":    "move_asset",
+					"space":     c.Prefix,
+					"slug":      appSlug,
+					"version":   version,
+					"filename":  filename,
+					"error_msg": err,
+				})
+				log.Error()
+			}
+		}()
 	}
+
+	content := bytes.NewReader(fileContent)
+	contentType = headers["Content-Type"]
+
 	att := &Attachment{
-		ContentType:   headers["Content-Type"],
-		Content:       bytes.NewReader(buf.Bytes()),
+		ContentType:   contentType,
+		Content:       content,
 		Etag:          headers["Etag"],
 		ContentLength: headers["Content-Length"],
 	}
 	return att, nil
 }
 
+// MoveToGlobalAssetDatabase moves an asset located in the "local" container in
+// the global database. This function is not intended to stay forever and will
+// be removed when no more assets will be remaining in the app containers.
+// It does the following steps:
+// 1. Creating the new asset object in the global database
+// 2. Adds a reference to the asset in the couch version document
+// 3. Removes the old asset location
+func MoveAssetToGlobalDatabase(c *Space, ver *Version, buf io.Reader, filename, contentType string) error {
+	globalFilepath := filepath.Join(c.Prefix, ver.Slug, ver.Version)
+
+	content, err := ioutil.ReadAll(buf)
+	if err != nil {
+		return err
+	}
+
+	h := md5.New()
+	_, err = h.Write(content)
+	if err != nil {
+		return err
+	}
+	md5Sum := h.Sum(nil)
+
+	a := &asset.GlobalAsset{
+		Name:        filename,
+		MD5:         hex.EncodeToString(md5Sum),
+		AppSlug:     ver.Slug,
+		ContentType: contentType,
+	}
+
+	err = asset.AssetStore.AddAsset(a, bytes.NewReader(content), globalFilepath)
+	if err != nil {
+		return err
+	}
+
+	// Updating the couch document
+	if ver.AttachmentReferences == nil {
+		ver.AttachmentReferences = make(map[string]string)
+	}
+	ver.AttachmentReferences[filename] = hex.EncodeToString(md5Sum)
+
+	db := c.VersDB()
+	_, err = db.Put(context.Background(), ver.ID, ver)
+	if err != nil {
+		return err
+	}
+	// Remove the old object
+	conf, err := config.GetConfig()
```

Comment:
We still should have a `config.GetConfig()` method that can't return an error (IIRC, it means moving swift init out of this method).